
syntax = "proto3";

package protobuf;

message Batch {

    message Simulate {

        message Experiment {
            string architecture = 1;
            string task = 2;

            // For grid accelerators
            uint32 n_lanes = 3;
            uint32 n_columns = 4;
            uint32 n_rows = 5;
            uint32 n_tiles = 6;
            uint32 column_registers = 7;
            uint32 bits_pe = 8;

            // For Pragmatic-Laconic
            bool booth_encoding = 9;
            uint32 bits_first_stage = 10;

            // For ShapeShifter-Loom
            uint32 group_size = 11;
            bool minor_bit = 12;

            // For Loom
            bool dynamic_weights = 13;
            uint32 pe_serial_bits = 14;

            // For Tactical
            uint32 lookahead_h = 15;
            uint32 lookaside_d = 16;
            string search_shape = 17;
            bool read_schedule = 18;

            // SCNN
            uint32 Wt = 19;
            uint32 Ht = 20;
            uint32 I = 21;
            uint32 F = 22;
            uint32 out_acc_size = 23;
            uint32 banks = 24;

            // On top architectures
            bool diffy = 25;
            bool tactical = 26;

        }

        uint32 batch = 1;
        uint32 epochs = 2;
        string model = 3;
        string data_type = 4;
        string network = 5;
        uint32 network_bits = 6;
        bool tensorflow_8b = 7;
        bool intel_inq = 8;
        repeated Experiment experiment = 9;
    }

    repeated Simulate simulate = 1;

}
